packages/Stntuple/package.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+import os
+
+def sanitize_environments(env, *vars):
+    for var in vars:
+        env.prune_duplicate_paths(var)
+        env.deprioritize_system_paths(var)
+
+class Stntuple(CMakePackage):
+    """Stntuple by Pavel Murat"""
+
+    homepage = "https://github.com/Mu2e/Stntuple"
+    git      = "https://github.com/Mu2e/Stntuple"
+    url      = "https://github.com/pavel1murat/frontends/archive/refs/tags/v1_04_00.tar.gz"  # not sure what it is...
+
+    license("BSD")
+
+    #    version("main", branch="main", get_full_repo=True)
+    #    version("v3_01_00", commit="e7b7abb733e00e8a97f31f02f87746fb29c4949e")
+#------------------------------------------------------------------------------
+# P.Murat: make sure we dont' update ELOG every time
+# elog-001.patch: fix (kludge) the installation directory
+#------------------------------------------------------------------------------
+    version("main"  , branch="main" , get_full_repo=True, submodules=True)
+    version("cmake" , branch="cmake", get_full_repo=True, submodules=True)
+    # patch("elog-001.patch")
+    def url_for_version(self, version):
+        url = "https://github.com/Mu2e/Stntuple/archive/refs/tags/{0}.tar.gz"
+        return url.format(version)
+
+    variant(
```

Comment:
I think you can consider just setting this in the cmake, since you won't generally be switching back and forth. Can then also remove cmake_flags.  I think the only case is if you want to support building one tag with multiple compiler versions, one of which doesn't support the preferred value.  If you want just in case, that's fine.